From 57bad816c710bd2f1b94bb201ebeaa30d64c597d Mon Sep 17 00:00:00 2001
From: Babbaj <babbaj45@gmail.com>
Date: Sun, 18 Jul 2021 18:13:59 -0400
Subject: [PATCH] allow sudo

---
 src/main.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/main.c b/src/main.c
index a94e1d4..bb724fa 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1097,11 +1097,11 @@ int main(int argc, char * argv[])
   // initialize for DEBUG_* macros
   debug_init();
 
-  if (getuid() == 0)
-  {
-    DEBUG_ERROR("Do not run looking glass as root!");
-    return -1;
-  }
+  //if (getuid() == 0)
+  //{
+  //  DEBUG_ERROR("Do not run looking glass as root!");
+  //  return -1;
+  //}
 
   DEBUG_INFO("Looking Glass (%s)", BUILD_VERSION);
   DEBUG_INFO("Locking Method: " LG_LOCK_MODE);
-- 
2.32.0

