diff --git a/client/renderers/EGL/texture_buffer.c b/client/renderers/EGL/texture_buffer.c
index cd596d10..bfe741ec 100644
--- a/client/renderers/EGL/texture_buffer.c
+++ b/client/renderers/EGL/texture_buffer.c
@@ -151,6 +151,7 @@ bool egl_texBufferStreamInit(EGL_Texture ** texture, EGL_TexType type,
 
   switch(type)
   {
+    case EGL_TEXTYPE_FRAMEBUFFER:
     case EGL_TEXTYPE_BUFFER_STREAM:
     case EGL_TEXTYPE_DMABUF:
       this->texCount = 2;
