diff --git a/pkgs/development/python-modules/openrazer/common.nix b/pkgs/development/python-modules/openrazer/common.nix
index c202ca374613..ec21ae37357e 100644
--- a/pkgs/development/python-modules/openrazer/common.nix
+++ b/pkgs/development/python-modules/openrazer/common.nix
@@ -4,10 +4,10 @@ rec {
   pyproject = true;
 
   src = fetchFromGitHub {
-    owner = "openrazer";
+    owner = "babbaj";
     repo = "openrazer";
-    tag = "v${version}";
-    hash = "sha256-pk3nghd16jhdf7zokwMzBGwGtBU7ta4nSHsOoGxjD4w=";
+    rev = "b46503e064605546ca40b6f8ad159e8ee2686794";
+    hash = "sha256-NMGEoLLJJhMroyZXV3sD/VIFxnWCre0JEJ7yQ6q8Tl8=";
   };
 
   meta = {
