diff --git a/pkgs/development/python-modules/openrazer/common.nix b/pkgs/development/python-modules/openrazer/common.nix
index 3e801b582797..7dd6c34f71c9 100644
--- a/pkgs/development/python-modules/openrazer/common.nix
+++ b/pkgs/development/python-modules/openrazer/common.nix
@@ -5,10 +5,10 @@
   format = "setuptools";
 
   src = fetchFromGitHub {
-    owner = "openrazer";
+    owner = "babbaj";
     repo = "openrazer";
-    rev = "v${version}";
-    hash = "sha256-tjVWvJxcZ2maR99VRwMGCa+IK+1CjCc7jxAj4XkDUEw=";
+    rev = "0408854d413d0c587b4fc07ab73a4707b75cf2f7";
+    hash = "sha256-Et3ctyqfYW5pKAm0JUFFOZ1Kgkb9guBUzL2iz3Rx35M=";
   };
 
   meta = with lib; {
